R/spaces_left_parentheses_linter.R

```diff
@@ -9,12 +9,15 @@ spaces_left_parentheses_linter <- function() {
 
         parsed <- source_file$parsed_content[id, ]
 
+        # nolint start: object_usage_linter.
+        # suppress "no visible binding" warnings in with() block
```

Comment:
do we also get the warning in `R CMD check`?

Reply:
`devtools::check()` was clean locally and in CI, so I guess not. Not sure why, though.

Reply:
Hmm it's because here:

https://github.com/jimhester/lintr/blob/25b2e1d430ba13b1436af4afd4332f27762f4f8b/R/pipe_continuation_linter.R#L76

Should `object_usage_linter` be seeing that too though?

(side note maybe it's better to put that call in a less hidden place than the bottom of a specific linter's source)

Reply:
We see those. Linted are `token`, `parent` and `terminator` IIRC.

You're right about moving the globalVariables call. Should be in `aaa.R` I think.

Reply:
Moved all globalVariables to `aaa.R`. As it turns out, `"from"` is not necessary as a global variable, so I removed that and instead added all `colnames()` of parsed content with a comment to explain what's going on.

Reply:
`with` is specifically skipped by `R CMD check` as near as I can tell, I believe it happens in this helper:

https://gitlab.com/luke-tierney/codetools/-/blob/master/R/codetools.R#L647-652

Anyway it's easy to write a simple package where `with` generates no notes.